src/main/java/moimoi/util/task/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package moimoi.util.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
+
+    /**
+     * Constructs a deadline task of the specified description and deadline.
+     *
+     * @param description Task description.
+     * @param by Task deadline.
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(TaskEnum.D, description);
+        this.by = by;
+    }
+
+    /**
+     * Checks if the task occurs on the specified date.
+     *
+     * @param date Date to be checked against the task's deadline.
+     * @return true if the task occurs on the specified date; false otherwise.
+     */
+    @Override
+    public boolean occursOn(LocalDate date) {
+        if (this.by.toLocalDate().isEqual(date)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```

Comment:
Great job with the occursOn method! I see you’ve implemented the date comparison effectively using isEqual. While your approach works perfectly, you could consider simplifying the method by returning the result of the condition directly, like this: `return this.by.toLocalDate().isEqual(date);` This way, the logic remains the same but the code becomes more concise. Of course, there’s no right or wrong here—just a suggestion for clarity. What do you think?